### PR TITLE
Update caml.py

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description**

1. Resolve deprecated warnings
- `nn.functional.tanh` -> `torch.tanh`
-  `set_deterministic` -> `use_deterministic_algorighms`
2. Update comments for CAML.

### CODE REVIEW

File: libmultilabel/networks/caml.py [hunk 1]
```diff
@@ -18,37 +17,39 @@ def __init__(self, config, embed_vecs):
...
         self.conv = nn.Conv1d(embed_vecs.shape[1], num_filter_per_size, kernel_size=filter_size, padding=int(floor(filter_size/2)))
         xavier_uniform_(self.conv.weight)
 
-        # Context vectors for computing attention as in 2.2
+        # Context vectors for computing attention
         self.U = nn.Linear(num_filter_per_size, config.num_classes)
         xavier_uniform_(self.U.weight)
 
-        # Final layer: create a matrix to use for the L binary classifiers as in 2.3
+        # Final layer: create a matrix to use for the L binary classifiers
```
Comment (cjlin1) [inline libmultilabel/networks/caml.py:28]:
```suggestion
        # Final layer: create a matrix to use for the #classes binary classifiers
```

File: libmultilabel/networks/caml.py [hunk 2]
```diff
@@ -18,37 +17,39 @@ def __init__(self, config, embed_vecs):
...
+        # alpha: (batch_size, label size, document length)
         alpha = torch.softmax(self.U.weight.matmul(x.transpose(1,2)), dim=2)
 
-        # Document representations are weighted sums using the attention. Can compute all at once as a matmul
+        # Document representations are weighted sums using the attention
+        # m: (batch_size, label size, num_filter_per_size)
         m = alpha.matmul(x)
 
-        # similarity
+        # Compute a probability for each label l in L
```
Comment (cjlin1) [inline libmultilabel/networks/caml.py:51]:
```suggestion
        # Compute a probability for each label
```

File: libmultilabel/utils.py [hunk 3]
```diff
@@ -91,7 +91,7 @@ def set_seed(seed):
     if seed is not None:
         if seed >= 0:
             seed_everything(seed=seed)
-            torch.set_deterministic(True)
+            torch.use_deterministic_algorithms(True)
```
Comment (cjlin1) [inline libmultilabel/utils.py:94]: So this is the new way to use deterministic settings?? I recall they issue a warning message?

**Review by cjlin1 — DISMISSED** [review]

File: libmultilabel/utils.py (hunk 3 above)
Comment (Eleven1Liu) [reply libmultilabel/utils.py:94]: Yes. They rename `set_deterministic` to `use_deterministic_algorithms ` for more clarity. 
https://github.com/pytorch/pytorch/issues/49100

**Review by cjlin1 — APPROVED** [review]

File: libmultilabel/utils.py (hunk 3 above)
Comment (Eleven1Liu) [reply libmultilabel/utils.py:94]: Note: `use_deterministic_algorithms` get error on `1.7.1+cu101`.

**Pull request merged**